AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.AzureWebApp
+{
+    using System;
+    using System.Net.Http;
+    using System.Runtime.Caching;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Class to contain the one cache for all Gauges 
+    /// </summary>
+    public class CacheHelper
+    {
+        /// <summary>
+        /// Checks if a key is in the cache and if not
+        /// Retrieves raw counter data from Environment Variables
+        /// Cleans raw JSON for only requested counter
+        /// Creates value for caching
+        /// </summary>
+        /// <param name="name">cache key and name of the counter to be selected from json</param>
+        /// <returns>value from cache</returns>
+        public static int GetCountervalue(string name)
+        {
+            if (!CacheHelper.IsInCache(name))
+            {
+                HttpClient client = new HttpClient();
+
+                /* http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ASPNET/
+                http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/
+                http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_CLR/
+               http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ALL/ */
+                Task<string> counterRetrieval = client.GetStringAsync("http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/");
+                counterRetrieval.Wait();
+
+                string json = counterRetrieval.Result;
+
+                string prefix = "\\\"" + name + "\\\":";
+                string postfix = ",\\";
+
+                int idx = json.IndexOf(prefix) + prefix.Length;
+                int endIdx = json.IndexOf(postfix, idx);
+
+                string value = json.Substring(idx, endIdx - idx);
+
+                CacheHelper.SaveToCache(name,value, DateTimeOffset.Now.AddSeconds(5.0));
```

Comment:
you need to save entire json into cache. Point here is that you want all counters to take the value from a single JSON taken at the same point in time